algorithms/src/snark/marlin/marlin.rs

```diff
@@ -727,11 +873,23 @@ where
 
         let mut evaluations = Evaluations::new();
 
+        let mut current_circuit_id = "".to_string();
+        let mut circuit_index: i64 = -1;
         for (label, (_point_name, q)) in query_set.to_set() {
             if AHPForR1CS::<E::Fr, MM>::LC_WITH_ZERO_EVAL.contains(&label.as_ref()) {
                 evaluations.insert((label, q), E::Fr::zero());
             } else {
-                let eval = proof.evaluations.get(&label).ok_or_else(|| AHPError::MissingEval(label.clone()))?;
+                if label != "g_1" {
+                    let circuit_id = format!("{}", CircuitId::from_witness_label(&label));
```

Comment:
```suggestion
                    let circuit_id = CircuitId::from_witness_label(&label).to_string();
```
`clippy` would eventually complain about this